### DISCUSSION TITLE
Fetch the span tags of the search phrase for custom modification

### DISCUSSION THREAD

**Discussion by rmdorsey** [stephanrauh/ngx-extended-pdf-viewer#2421] [Q&A]
Hi, thanks for the great project!

The programmatic find functionality works great, but I'm wondering if there's a version that doesn't presume to highlight, but returns the span tags (and their content including the inner-text) that get the highlight class added to them if they're part of the phrase being searched for.

**The reasons for my ask:**
Some words are common throughout a large PDF.  I have a corpus of keywords I'd like highlighted but only if they are within the phrase being searched for.  Once the functionality I described in my previous paragraph occurs I'd then want to search through theresults, find the keyword(s), and finally apply the css class to highlight.  

The other reason for my ask is because I'd like to apply my own custom css class names to the relevant span tags - that way the highlight will remain even if a user wants to search for other words/phrases.  Current functionality removes highlights once a new search is performed.

Thanks!
Matt

**Comment by stephanrauh** [comment]
Have a look at my "export lines" implementation: https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/27da4a7649e978aea58ee87433acfe6aba55bfac/projects/ngx-extended-pdf-viewer/src/lib/ngx-extended-pdf-viewer.service.ts#L202-L271

I can't remember wether this code has access to the actual `<span>` tags - but I think it does.

**Reply by rmdorsey** [reply]
Thanks!  Is there code that can also do the search for me?  Finding phrases in some PDFs is proving to be very difficult because words are broken up and spread across two or more span tags.  I see that your code above narrows things down to a specific page.  After that I'd want to narrow it down to the search phrase and then after that I'd want to search for the specific key words within the search phrase.

Thanks!
